File: C++/2-Medium/00259._较小的三数之和/00259._较小的三数之和.cpp
#include <iostream>
#include <vector>
#include <algorithm> // 用于sort函数
using namespace std;

// 定义三数之和小于target的函数
int threeSumSmaller(vector<int>& nums, int target) {
    sort(nums.begin(), nums.end()); // 对数组进行排序
    int count = 0; // 用于计数满足条件的三元组数量
    int n = nums.size();
    // 遍历数组
    for (int i = 0; i < n - 2; ++i) {
        int left = i + 1, right = n - 1; // 设置左右指针
        while (left < right) { // 当左指针小于右指针时循环
            int sum = nums[i] + nums[left] + nums[right];
            if (sum < target) { // 如果三数之和小于target
                count += right - left; // 更新计数，right-left是因为如果a+b+c<target，则a+b+(c到right之间的任意数)都满足条件
                ++left; // 移动左指针
            } else {
                --right; // 移动右指针以减小和
            }
        }
    }
    return count; // 返回满足条件的三元组数量
}

int main() {
    vector<int> nums = {-2, 0, 1, 3}; // 示例数组
    int target = 2; // 目标值
    cout << "满足条件的三元组数量为: " << threeSumSmaller(nums, target) << endl;
    return 0;
}

File: Python/2-Medium/00259._较小的三数之和/00259._较小的三数之和.py
from typing import List

def three_sum_smaller(nums: List[int], target: int) -> int:
    """
    计算数组中三个数之和小于目标值的组合数量。

    Args:
    nums: List[int] - 整数数组。
    target: int - 目标值。

    Returns:
    int - 组合数量。

    示例:
    >>> three_sum_smaller([-2, 0, 1, 3], 2)
    2
    """
    nums.sort()  # 对数组进行排序
    count = 0  # 用于计数满足条件的组合数
    n = len(nums)
    
    for i in range(n-2):  # 固定第一个数，遍历到倒数第三个即可
        left, right = i+1, n-1  # 双指针初始化
        while left < right:
            if nums[i] + nums[left] + nums[right] < target:
                # 当前固定数和左右指针指向的数之和小于目标值
                count += right - left  # 加上所有可能的组合数
                left += 1  # 移动左指针以探索新的可能性
            else:
                right -= 1  # 和过大，需要移动右指针减小和

    return count

# 测试函数
test_nums = [-2, 0, 1, 3]
test_target = 2
result = three_sum_smaller(test_nums, test_target)
print(result)